src/DataStructures/VectorImpl.hpp

```diff
@@ -119,12 +128,12 @@ struct MarkAsVectorImpl {};
  *  vector types in math expressions.
  *
  * \note
- * - If created with size 0, then  `data()` will return `nullptr`
  * - If either `SPECTRE_DEBUG` or `SPECTRE_NAN_INIT` are defined, then the
  *   `VectorImpl` is default initialized to `signaling_NaN()`. Otherwise, the
  *   vector is filled with uninitialized memory for performance.
  */
-template <typename T, typename VectorType>
+template <typename T, typename VectorType,
+          size_t StaticSize = default_vector_impl_static_size>
 class VectorImpl
```

Comment:
The TensorExpression code, at least, assumes `VectorImpl` only has two parameters.  For example, this won't match custom-sized vectors:
```c++
template <typename ValueType, typename VectorType>
struct get_binop_datatype_impl<VectorImpl<ValueType, VectorType>, ValueType> {
```